### PR TITLE
[BPK-1683] stories consistency

### PR DESCRIPTION
**Pull request by matteo-hertel**
This is the first of 3 PRs, it focuses on 2 main things:
1. only the story that will be added to the docs site have the prefix `docs:` and I've capitalised some names here and there
2. fix some docs issues, components not appearing in the sidebar and redirects

### CODE REVIEW

File: native/packages/react-native-bpk-component-banner-alert/stories.js [hunk 1]
```diff
@@ -166,7 +166,7 @@ class BpkBannerAlertFadeDemo extends Component<
 
 storiesOf('react-native-bpk-component-banner-alert', module)
   .addDecorator(CenterDecorator)
-  .add('docs:banner-alerts', () => (
+  .add('Banner alerts', () => (
```
Comment (matthewdavidson) [inline native/packages/react-native-bpk-component-banner-alert/stories.js]: It looks like you are removing the `docs:default` example here but on further inspection you are not ... perhaps moving the `docs:` stories to the top will help?

File: packages/bpk-docs/src/constants/redirect-routes.js [hunk 2]
```diff
@@ -60,7 +60,8 @@ export default {
   '/components/native/badge': ROUTES.BADGE,
   '/components/native/banner-alerts': ROUTES.BANNER_ALERT,
   '/components/native/button': ROUTES.BUTTON,
-  '/components/native/button-link': ROUTES.BUTTON_LINK,
```
Comment (matthewdavidson) [inline packages/bpk-docs/src/constants/redirect-routes.js]: Whats happening here?

File: native/packages/react-native-bpk-component-banner-alert/stories.js (hunk 1 above)
Comment (matteo-hertel) [reply native/packages/react-native-bpk-component-banner-alert/stories.js]: yes absolutely, good catch!

on a side note, why is always banner alert creating problems? 😞

File: packages/bpk-docs/src/constants/redirect-routes.js (hunk 2 above)
Comment (matteo-hertel) [reply packages/bpk-docs/src/constants/redirect-routes.js]: `/components/native/button-link` goes to `/components/link`
same for 
'/components/button-link' as we call em just link across the board, the component name in RN is misleading

**Comment by shaundon** [comment]
Checked through and I think it's good. The build failed but looks like it was terminated, so I've restarted it. Once it passes I'll ✅

**Pull request merged**